### ISSUE TITLE
VS Code 1.66 requires GLIBC 2.25 on RHEL 7

### ISSUE THREAD

**Issue by bddali** [microsoft/vscode#146390]
VS Code version: code-1.66.0-1648620680
OS: Red Hat Enterprise Linux 7.9

Running yum update on RHEL 7 produces an error:

Error: Package: code-1.66.0-1648620680.el7.x86_64 (code)
           Requires: libc.so.6(GLIBC_2.25)(64bit)

According to https://code.visualstudio.com/docs/supporting/requirements the required version of GLIBC is 2.15 or later.

Red Hat Enterprise Linux 7 uses GLIBC 2.17.

**Comment by dirtyharrycallahan** [comment]
Just ran into this when trying to upgrade this moring.

**Comment by cmille62** [comment]
I also ran into this problem trying to install today

**Comment by bpasero** [comment]
cc @deepak1556 @rzhao271

**Comment by markusdd** [comment]
can confirm as well.

There finally needs to be a pipeline that checks RHEL/CentOS 7 comptability. This is a very important stable developer OS with support until 2024.

We went through this with the false electron update a while back and now this silently broke again.

**Comment by kevinbwusa** [comment]
Same problem here: 
yum update -y code                                                                                                                                                                                                                                                   50.54 Dur  18:39:25
Loaded plugins: enabled_repos_upload, langpacks, package_upload, product-id, search-disabled-repos, subscription-manager
rhel-7-server-extras-rpms                                                                                                                                                                                                                                                                              | 2.0 kB  00:00:00     
rhel-7-server-optional-rpms                                                                                                                                                                                                                                                                            | 2.0 kB  00:00:00     
rhel-7-server-rpms                                                                                                                                                                                                                                                                                     | 2.0 kB  00:00:00     
rhel-7-server-satellite-tools-6.7-rpms                                                                                                                                                                                                                                                                 | 2.1 kB  00:00:00     
rhel-server-rhscl-7-rpms                                                                                                                                                                                                                                                                               | 2.0 kB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package code.x86_64 0:1.65.2-1646927812.el7 will be updated
---> Package code.x86_64 0:1.66.0-1648620680.el7 will be an update
--> Processing Dependency: libc.so.6(GLIBC_2.25)(64bit) for package: code-1.66.0-1648620680.el7.x86_64
--> Finished Dependency Resolution
Error: Package: code-1.66.0-1648620680.el7.x86_64 (code)
           Requires: libc.so.6(GLIBC_2.25)(64bit)
**********************************************************************
Dependency resolving failed due to missing dependencies.
Some repositories on your system are disabled, but yum can enable them
and search for missing dependencies. This will require downloading
metadata for disabled repositories and may take some time and traffic.
**********************************************************************

rhel-7-server-supplementary-rpms/x86_64                                                                                                                                                                                                                                                                | 2.0 kB  00:00:00     
rhel-7-server-ansible-2-rpms/x86_64                                                                                                                                                                                                                                                                    | 2.4 kB  00:00:00     
rhel-7-server-satellite-tools-6.6-rpms/x86_64                                                                                                                                                                                                                                                          | 2.1 kB  00:00:00     
rhel-7-server-rh-common-rpms/x86_64                                                                                                                                                                                                                                                                    | 2.1 kB  00:00:00     
--> Running transaction check
---> Package code.x86_64 0:1.66.0-1648620680.el7 will be an update
--> Processing Dependency: libc.so.6(GLIBC_2.25)(64bit) for package: code-1.66.0-1648620680.el7.x86_64
--> Finished Dependency Resolution
Error: Package: code-1.66.0-1648620680.el7.x86_64 (code)
           Requires: libc.so.6(GLIBC_2.25)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
Uploading Enabled Repositories Report
Loaded plugins: langpacks, product-id, subscription-manager
Time: 0h:00m:11s

**Comment by chriscasola** [comment]
It it possible this issue would also cause the vscode remote server to fail to start inside a rhel 7 container? I can no longer use the "clone in container" option with a project using a rhel7 base image.

**Comment by deepak1556** [comment]
@chriscasola it is unrelated, please open a separate issue for it at https://github.com/microsoft/vscode-remote-release/issues to investigate further, thanks!

**Comment by deepak1556** [comment]
Just an update on the issue, it is not caused by electron update. Rather with `1.66` we automated the process of how the package dependencies are generated as part of our build pipeline https://github.com/microsoft/vscode/issues/17142, which now generates the right set of dependencies for all the native modules and binaries shipped with the package. The current `GLIBC_2.25` requirement was generated from one our native module dependency and this modules dependency has not changed for a while now, basically we were previously shipping with an incorrect dependency list which was now caught.  

Although we had a test plan item for verifying this new process https://github.com/microsoft/vscode/issues/145608, we had missed to verify on CentOS 7. Sorry about the update breakage!

**Comment by markusdd** [comment]
thank you for the update! Is there an ETA for a fix?

This is currently causing a lot of stir because regular updates now fail without the --skip-broken option.

**Comment by deepak1556** [comment]
Verification steps on CentOS 7:

* Download latest rpm from https://code.visualstudio.com/insiders/#linux
* `rpm -ivh <rpm-file>`
* Launch the application and login to settings sync
* Ensure the credentials are encrypted in the system keystore for key starting with `vscode-insiders-`

**Comment by cmille62** [comment]
This still does not allow installation: 

`rpm -ivh code-insiders-1.67.0-1649050674.el7.x86_64.rpm 
error: Failed dependencies:
	libc.so.6(GLIBC_2.25)(64bit) is needed by code-insiders-1.67.0-1649050674.el7.x86_64`

`sudo yum install code
Loaded plugins: langpacks, versionlock
Resolving Dependencies
--> Running transaction check
---> Package code.x86_64 0:1.66.0-1648620680.el7 will be installed
--> Processing Dependency: libc.so.6(GLIBC_2.25)(64bit) for package: code-1.66.0-1648620680.el7.x86_64
--> Finished Dependency Resolution
Error: Package: code-1.66.0-1648620680.el7.x86_64 (code)
           Requires: libc.so.6(GLIBC_2.25)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
`

**Comment by deepak1556** [comment]
Insiders with the fix has not been released yet, it will be notified in this thread once the builds are available.

**Comment by imokech** [comment]
I have the same problem and I could not install VSCode 1.66 on centos 7.

**Comment by ajkerzner** [comment]
Looks good to me, RHEL 7.9 with the insider build via standard install.
Workaround: `rpm -Uvh --nodeps <vscode rpm>`.

**Comment by bddali** [comment]
Succesfully installed code-insiders-1.67.0-1649156474.el7.x86_64.rpm on RHEL 7.

/verified

**Comment by selwynchambersmetoffice** [comment]
@deepak1556 so is this issue fixed?

**Comment by paulds1** [comment]
I'm not familiar with vscode release procedures... What is the timeline for getting the fixed build pushed to Stable?  Just wondering when we should expect our production systems to see the update.

**Comment by isidorn** [comment]
To learn more about our recovery releases please read this wiki post https://github.com/microsoft/vscode/wiki/Running-the-Endgame#recovery-build
To understand our development process better please check this out https://github.com/microsoft/vscode/wiki/Development-Process

This fix should be in stable either tomorrow, or early next week. So in the next 6 days.
In the meantime please use the vscode insiders build if you are blocked, it has the fix. Thanks
